docs/contributing/development.md

```diff
@@ -1,62 +1,54 @@
-# Development environment setup
+# Development
+
+## Getting the code
 
 1. Fork [nsidc/earthaccess](https://github.com/nsidc/earthaccess)
 1. Clone your fork (`git clone git@github.com:{my-username}/earthaccess`)
 
-`earthaccess` uses Poetry to build and publish the package to PyPI, the defacto Python repository. In order to develop new features or fix bugs etc. we need to set up a virtual environment and install the library locally. We can accomplish this with Conda and Poetry, or just with Poetry.  Both workflows achieve the same result.
+In order to develop new features or fix bugs etc. we need to set up a virtual
+environment and install the library locally.
 
-### Using Conda
+## Quickstart development
 
-If we have `mamba` (or `conda`) installed, we can use the environment file included in the `ci` folder. This will install all the libraries we need (including Poetry) to start developing `earthaccess`:
+The fastest way to start with development is to use nox. If you don't have nox,
+you can use `pipx run nox` to run it without installing, or `pipx install nox`.
+If you don't have pipx (pip for applications), then you can install with
+`pip install pipx` (the only case were installing an application with regular
+pip is reasonable). If you use macOS, then pipx and nox are both in brew, use
+`brew install pipx nox`.
 
-```bash
-mamba env update -f ci/environment-dev.yml
-mamba activate earthaccess-dev
-poetry install
+To use, run `nox`. This will typecheck and test using every installed version of
+Python on your system, skipping ones that are not installed. You can also run
+specific jobs:
+
+```console
+$ nox -s typecheck              # Typecheck only
+$ nox -s tests                  # Python tests
+$ nox -s build_docs -- --serve  # Build and serve the docs
+$ nox -s build_pkg              # Make an SDist and wheel
 ```
 
-After activating our environment and installing the library with Poetry we can run Jupyter lab and start testing the local distribution or we can use `make` to run the tests and lint the code.
-Now we can create a feature branch and push those changes to our fork!
+Nox handles everything for you, including setting up a temporary virtual
+environment for each run.
 
-### Using Poetry
+## Manual development environment setup
 
-If we want to use Poetry, first we need to [install it](https://python-poetry.org/docs/#installation). After installing Poetry we can use the same workflow we used for Conda, first we install the library locally:
+Create and activate a virtual environment with `venv`, which comes by default with
+Python, in the `.venv` directory:
 
 ```bash
-poetry install
+python -m venv .venv
+source .venv/bin/activate
 ```
 
-and now we can run the local Jupyter Lab and run the scripts etc. using Poetry:
+Install _earthaccess_ in editable mode with optional development dependencies:
 
 ```bash
-poetry run jupyter lab
+pip install --editable .[dev,test,docs]
```

Comment:
Can we also expose this as a nox command instead?

Reply:
I'm not sure! What would this look like? It doesn't really map to how I normally use and think about `nox`. In my mental model I create "sessions" as independent and stateless things, e.g. running typechecking nox creates and can cache an environment with mypy and all the runtime dependencies, and for linting, nox can depend on an environment that solely contains ruff.

Maybe this is more common than I think, but using nox to do setup just doesn't seem natural to me :)

Reply:
I consider `nox` analogous to `make`, both of which I consider being able to wrap "complex" commands within simpler commands. So, rather than having to type the `pip` command above, someone might instead be able to type `nox install` and be done with it, without fat-fingering arguments.

To take this further, `nox` (and `make` et al.) can eliminate duplication. Another suggestion I have is to modify the github workflows to _also_ use `nox` rather than the "bare" commands. This way, both a contributor and github are using the same commands (both using `nox`). Granted, a contributor might not run all of the things that github would, but by having both use `nox` (in our case), there's no need to update docs or workflows to match changes to the noxfile (for _existing_ commands), thus eliminating (or greatly reducing) the likelihood of docs and files getting out of sync.

However, to expedite landing this PR, we can open a separate issue to address this.

Reply:
I agree that this should move into a separate PR. How we think about tasks runners, philosophically, is complicated :) This page in the scientific python library development guide provides some suggestions: https://learn.scientific-python.org/development/guides/tasks/#task-runners